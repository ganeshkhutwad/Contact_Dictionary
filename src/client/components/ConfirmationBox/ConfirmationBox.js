import React from 'react';

const ConfirmationBox = ({ value, deleteRecord }) => {
    return (
        <div>
            <div> Do you want to delete contact with id = {value}</div>
            <button onClick={deleteRecord}>Ok</button>
        </div>
    )
}

export default ConfirmationBox;