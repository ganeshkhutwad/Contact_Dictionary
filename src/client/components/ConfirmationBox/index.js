export { default as ConfirmationBox } from './ConfirmationBox';