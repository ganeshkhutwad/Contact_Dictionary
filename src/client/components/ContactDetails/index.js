export { default as ContactDetails } from './ContactDetails';