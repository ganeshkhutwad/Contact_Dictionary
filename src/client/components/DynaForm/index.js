export { default as DynaForm } from './DynaForm';
