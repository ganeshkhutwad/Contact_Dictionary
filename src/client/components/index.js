export { Header } from './Header';
export { ContactList } from './ContactList';
export { ContactDetails } from './ContactDetails';
export { AddButton } from './AddButton';
