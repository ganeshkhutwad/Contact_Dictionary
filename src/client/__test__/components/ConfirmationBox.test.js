/**
Test `AddButton` Presentational Component.
@author Ganesh Khutwad
 */
import React from 'react';
import { ConfirmationBox } from '../../components/ConfirmationBox';

const value = 10;
const deleteRecord = () => { }

describe('`ConfirmationBox` Tests', () => {
    it('`ConfirmationBox` Snapshot', () => {
        const confirmationBoxWrapper = shallow(<ConfirmationBox value={value} deleteRecord={deleteRecord} />);
        expect(confirmationBoxWrapper).toMatchSnapshot();
    });
});
