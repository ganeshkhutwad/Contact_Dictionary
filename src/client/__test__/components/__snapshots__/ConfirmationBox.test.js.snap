// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`ConfirmationBox\` Tests \`ConfirmationBox\` Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(ConfirmationBox)
    deleteRecord={[Function]}
    value={10}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "okBtn": "ConfirmationBox-okBtn-1",
      },
      "deleteRecord": [Function],
      "value": 10,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "okBtn": "ConfirmationBox-okBtn-1",
        },
        "deleteRecord": [Function],
        "value": 10,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
